### ISSUE TITLE
Date support

### ISSUE THREAD

**Issue by ciscoheat** [gcornut/valibot-json-schema#3]
Hey, thank you for this library, it's exactly what's needed for [Superforms v2](https://superforms.rocks/migration-v2/)!

I'm curious about your thoughts for the date validator. I'm using a couple of other "toJSONSchema" validators for other libraries (Joi, Zod, TypeBox, etc) and they have a slightly different strictness of the type they generate for the schema.

TypeBox is using `type: 'Date'`, which is not a part of the specification but still understandable. Most runtimes parse a timestamp or a string into a Date instance, so they leave the implementation to the library consumer.

[zod-to-json-schema](https://github.com/StefanTerdell/zod-to-json-schema#readme) has an option for generating one of these schemas:

Strategy "integer"
```ts
{ type: 'integer', format: 'unix-time' }
```

Strategy "string"
```ts
{ type: 'string', format: 'date-time' }
```

Where I have decided to prefer the "integer" strategy, since it's more common that a datetime string should actually be kept as a string. If you specify an integer in unix-time format as a default value in the schema (for a date field, to be precise), it will be converted to a `Date` instance.

What's your take on this? I'm happy to contribute a PR.

**Comment by gcornut** [comment]
Hey, thanks for the feedback !

It made me think I did not have the support for string `format: "date-time"` converted from `string([isoTimestamp()])` (I just pushed an update). 

I would be open to adding the conversion of valibot `date()` schema to JSON schema string or integer as an option but it's probably best not to convert it by default to force the user to think about how they want to represent date in JSON. 

If you want, you can definitely create a PR

**Comment by ciscoheat** [comment]
Ok nice, I'll make a PR for that. Also working on a default values PR, will submit that one separately.

**Comment by ciscoheat** [comment]
PR added: #5

**Comment by gcornut** [comment]
Merged and publish in v0.0.21
